extensions/ql-vscode/src/vscode-tests/ensureCli.ts

```diff
@@ -0,0 +1,129 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { DistributionManager, extractZipArchive, codeQlLauncherName } from '../distribution';
+import fetch from 'node-fetch';
+
+/**
+ * This module ensures that the proper CLI is available for tests of the extension.
+ * There are two environment variables to control this module:
+ *
+ *     - CLI_VERSION: The version of the CLI to install. Defaults to the most recent
+ *       version. Note that for now, we must maintain the default version by hand.
+ *
+ *     - CLI_BASE_DIR: The base dir where the CLI will be downloaded and unzipped.
+ *       The download location is `${CLI_BASE_DIR}/assets` and the unzip loction is
+ *       `${CLI_BASE_DIR}/${CLI_VERSION}`
+ *
+ * After downloading and unzipping, a new environment variable is set:
+ *
+ *     - CLI_PATH: Points to the cli executable for the specified CLI_VERSION. This
+ *       is variable is available in the unit tests and will be used as the value
+ *       for `codeQL.cli.executablePath`.
+ *
+ * As an optimization, the cli will not be unzipped again if the executable already
+ * exists. And the cli will not be re-downloaded if the zip already exists.
+ */
+
+process.on('unhandledRejection', e => {
+  console.error('Unhandled rejection.');
+  console.error(e);
+  process.exit(-1);
+});
+
+const _1MB = 1024 * 1024;
+const _10MB = _1MB * 10;
+
+// CLI version to test. Hard code the latest as default. And be sure
+// to update the env if it is not otherwise set.
+const CLI_VERSION = process.env.CLI_VERSION || 'v2.3.1';
```

Comment:
I think you can hardcode `latest`, though you should check the API docs to make sure the asset download URL still works the same.

Reply:
Unfortunately,

This does not work:
https://github.com/github/codeql-cli-binaries/releases/download/latest/codeql-osx64.zip

But this does:
https://github.com/github/codeql-cli-binaries/releases/download/v2.3.1/codeql-osx64.zip

Reply:
Also, a downside of using `latest` locally is that once you download `latest` once, you will never re-download. We check whether or not to re-download based on the presence of the directory name. The directory name is generated based on the version number. I could probably be more sophisticated about this, but that would add complexity.